src/Microsoft.TestPlatform.CoreUtilities/Extensions/GuidPolyfill.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System;
+/// <summary>
+/// A polyfill helper for Guid.
+/// </summary>
+internal static class GuidPolyfill
```

Comment:
@MarcoRossignoli Was suggesting to have only one class called `Polyfill` that would host all polyfills as sub-classes (e.g. `Polyfill.Guid.Parse`. It looks like a good idea but doesn't allow for extension methods (need to be defined at top class-level).

We can:
1. Keep solution implemented in this PR
2. Go with @MarcoRossignoli's suggestion for non-extension methods (not ideal as it would force us to have 2 ways)
3. Same as 2. but we do that also for extension methods (will no longer be extension methods).

Reply:
I see theoretical way around it but it requires three classes and instances. Seems like PolyfillGuid or GuidPolyfill in separate class is much easier way to go.

Reply:
it's ok as-is to me.